### PR TITLE
Update to version 4.7.1 and dependencies

### PR DESCRIPTION
**Pull request by corecontingency**
I really hope this works!

### CODE REVIEW

**Comment by corecontingency** [comment]
Apparently, module `glibmm` requires `libsigc++` version 2 to be built, and it complains about missing dependencies and exits the build with `libsigc++` version 3. Reverting.

**Comment by corecontingency** [comment]
How funny! Now module `cairomm` is complaining about not having `libsigc++` version three! I will see if I cannot add both versions of `libsigc++`.

**Comment by wwmm** [comment]
PulseEffects uses `libsigc++ 2.0`. I think that only when GTK4 is released a port to version 3 will make sense. So I would not bother with it now.

In case it helps these are the versions I have in Arch Linux:
```
libsigc++ 2.10.2
cairomm 1.12.2
gtkmm3 3.24.2
```

**Comment by corecontingency** [comment]
So the build fails when trying to apply the `zita-convolver` patch. I am not sure what is broken.
```
========================================================================
Building module zita-convolver in /srv/buildbot/worker/build-x86_64-6/build/.flatpak-builder/build/zita-convolver-1
========================================================================
]2;flatpak-builder: Building zita-convolverFB: Running: tar xf /srv/buildbot/worker/build-x86_64-6/build/.flatpak-builder/downloads/62d7841757f10c094e43ed755e187f947c5743f302ed2a1ee6064a850c18921466f4505d8a2a7b3ad23619db7f1ad7307e1dfb2e8a1e7685e60ece2ffff4f6ca/zita-convolver-4.0.3.tar.bz2 --no-same-owner --strip-components=1 -j
Applying patch 0001-Fix-makefile.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64-6/build/patch/zita-convolver/0001-Fix-makefile.patch
patching file source/Makefile
]2;flatpak-builder: Cleanup zita-convolverError: module zita-convolver: Can't find makefile
FB: unmounting rofiles-fuse /srv/buildbot/worker/build-x86_64-6/build/.flatpak-builder/rofiles/rofiles-EM86nx
program finished with exit code 1
elapsedTime=845.616416
```

**Comment by wwmm** [comment]
What does this patch do? Arch Package does not seem to do anything unusual to compile it https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/zita-convolver

**Comment by corecontingency** [comment]
So I don't have a very good understanding about this, but I will do my best to break it down. I mostly tried to keep things the same as before to try to minimize breakage. The patch seems to do three things.

1. Remove `march=native`
2. Remove `ldconfig`
3. Add missing symlink for major version of lib

Arch has this in code in the PKGBUILD:
```
  # FS#36248 - strip -march=native
  # call ldconfig in local context
  sed -e '/native/d' \
      -e 's/ldconfig/& -N $(DESTDIR)\/$(LIBDIR)/' \
      -i source/Makefile
```
So the PKGBUILD does the same thing as the first item of the patch, removing `march=native`. `march=native` seems to be a [configuration option passed to the compiler that tells it to use optimization settings based on whatever CPU architecture that is being used.](https://stackoverflow.com/questions/3015306/what-exactly-does-march-native-do)

Regarding the second item, while the patch deletes `ldconfig` entirely, the arch PKGBUILD only changes its properties, telling it to run in a local context, and passing the configuration option -N. [`ldconfig` seems to create the necessary links and cache to the most recent shared libraries](https://linux.die.net/man/8/ldconfig), while option -N tells it to not rebuild the system cache.

The arch PKGBUILD does not seem to do the third item the patch does (Add missing symlink for major version of lib). I am not sure what the reasoning behind this third item is.

Maybe the patch should be changed to be more in-line with the arch PKGBUILD? However, I do not think that is the cause of the flatpak build failing. This line implies that the program cannot find the Makefile to patch for some reason, causing the problem.

`]2;flatpak-builder: Cleanup zita-convolverError: module zita-convolver: Can't find makefile`

**Comment by wwmm** [comment]
Removing `-march=native` is necessary. Otherwise the package may not work on different CPUs. I saw the `Can't find makefile` message but I was wondering if the patch was touching it. IF that is not the case there are a few questions that may help
```
Could the zita source download be being corrupted somehow? From where are you downloading it?
Does it come with this makefile in its sources or is this makefile automatically generated at build time? If it is generated it could be a write permission problem somewhere where zita scripts need to write this makefile.
```

**Comment by wwmm** [comment]
I downloaded the zita source. It comes with the Makefile... Weird...

**Comment by wwmm** [comment]
The error line also shows `Cleanup`. Could it be trying to run `make clean` and failing because there is no permission to run `/bin/rm -f *~ *.o *.a *.d *.so.*` as requested in zita makefile? Is this cleaning phase necessary? Maybe you could just skip it.

**Comment by corecontingency** [comment]
Same error as before:
```
========================================================================
Building module zita-convolver in /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/build/zita-convolver-1
========================================================================
]2;flatpak-builder: Building zita-convolverFB: Running: tar xf /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/downloads/62d7841757f10c094e43ed755e187f947c5743f302ed2a1ee6064a850c18921466f4505d8a2a7b3ad23619db7f1ad7307e1dfb2e8a1e7685e60ece2ffff4f6ca/zita-convolver-4.0.3.tar.bz2 --no-same-owner --strip-components=1 -j
Applying patch 0001-Fix-makefile.patch
FB: Running: patch -p1 -i /srv/buildbot/worker/build-x86_64/build/patch/zita-convolver/0001-Fix-makefile.patch
patching file source/Makefile
]2;flatpak-builder: Cleanup zita-convolverError: module zita-convolver: Can't find makefile
FB: unmounting rofiles-fuse /srv/buildbot/worker/build-x86_64/build/.flatpak-builder/rofiles/rofiles-uuQmMz
program finished with exit code 1
elapsedTime=847.240260
```
Very confused at what is broken. Maybe it is looking for a file named `makefile`, and cannot find it because the file is named `Makefile` instead? But that doesn't really make sense, because the file was named Makefile in the previous version of zita...

**Comment by wwmm** [comment]
I downloaded the zita source and both `make` and `make clean` ran fine without any need to do anything. Whatever it is happening is related to the building inside flatpak. But I have no idea of what  could be causing such an error.

What happens if you do not apply the patch? The patch is necessary but maybe it should be applied in a different way.

**Comment by AsavarTzeth** [comment]
If my memory serves me I originally pulled the zita patch from Arch Linux, but it has likely changed since then. I also added a symlink that was needed for the flatpak. See commits:
- 3a4707b17ea55b84637cd4c62c1d92396fa6f756
- 3dfa80710a149c6e715de255b9da620875887ef3

Usually I do a much better job at documenting all changes, but in this case it was admittedly a bit vague.

I will see if I can figure it out. Good work with the rest!

**Comment by wwmm** [comment]
Looking at the entire log I saw `Library zita-convolver found: NO`. It seems that zita is not being found. This is making the convolver plugin compilation fail `FAILED: src/convolver/libgstpeconvolver.so `. This is probably the reason why the build was aborted.

It is good to know that Boost is going to deprecate a header. Are you using version 1.72? At least on version 1.71 I do not see this warning. In any case I will move to the new header before releasing the next version.

**Comment by AsavarTzeth** [comment]
Boost 1.71 is currently used, but it is updated to 1.72 as a part of this pull request.

Looking at the log it would seem zita-convolver is built correctly, but the library files are for some reason installed in the source dir, rather than `/app/lib`.

**Comment by corecontingency** [comment]
The makefile is significantly different for zita-convolver 4.0.3 compared to the old version of 4.0.0. Let's see if this works!

**Comment by wwmm** [comment]
Same problem `Library zita-convolver found: NO`

File: com.github.wwmm.pulseeffects.yml [hunk 1]
```diff
@@ -493,7 +491,7 @@ modules:
               - LIBDIR=lib
```
Comment (AsavarTzeth) [inline com.github.wwmm.pulseeffects.yml]: I found the final zita-convolver issue. The update to 4.0.3 changed the behaviour of the LIBDIR variable to instead expect the absolute path (it now includes the PREFIX).

**Comment by AsavarTzeth** [comment]
You may revert the last 2 commits, they are unnecessary (5aca6db083947c9953dfdb818d978daac7e20d18 and b395c2aa0f052490780af337610f6ed0c47d2ec8).

**Comment by corecontingency** [comment]
Sorry about that. This commit history is getting to be a bit of a nightmare, should I git rebase before merging? 

I think rebasing is what I want; I am a little new to git/version control.

**Comment by corecontingency** [comment]
bot, build

**Comment by corecontingency** [comment]
Looks good! Tried it out and everything seems to work fine. Running Arch with GNOME on Wayland.

**Pull request merged**